### ISSUE TITLE
Official 7" Raspberry Touch Screen together with ivport dual v2

### ISSUE THREAD

**Issue by Quas7** [ivmech/ivport#28]
Hi,

I have trouble using the 7" touchscreen together with the IVport.
As soon as display and IVport are connected I get ghosts i2c devices shown in i2cdetect.
I already tried to lower the i2c speed from 100 kHz down to 10 kHz but that did not change anything.
--> I suspect a noise problem on the i2c bus.

Next step: I will get an oszi and check the bus for noise because of the display. Maybe I need smaller  pullup resistors for the bus.

If anybody has already solved this issue, please comment.

Best regards
Quas7

Some more details...

I use the following setup:
Raspberry Pi 3 Model B, Raspberrian 4.4.38-v7
IVport dual v2: Pi NoIR Cam V2, Pi Cam V2, jumper "1" to connect GPIO 11 to the IVport
Both, display and raspberry are powered by 5V, 3A supplies - no "undervoltage" can be seen.

As long as I do not attach the display, I get the following "i2cdetect -y 1" output:

     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- --
10: 10 -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- 64 -- -- -- -- -- -- -- -- -- -- --
70: 70 -- -- -- -- -- -- --

10, 64, 70 are expected to be there and everything is working

But, after connection the display (just the flat ribbon cable, no power) I get random device addresses on the I2C bus:

     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- --
10: 10 -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- 38 -- -- -- -- -- -- --
40: -- -- -- -- -- 45 -- -- -- 49 4a -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- 64 -- -- -- -- -- -- -- -- -- -- --
70: 70 -- -- -- -- -- -- --

or... 

     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- --
10: 10 -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- 38 -- -- 3b 3c 3d -- --
40: -- -- -- -- -- 45 -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- 64 -- -- -- -- -- -- -- -- -- -- --
70: 70 -- -- -- -- -- -- --

Booting only the display without the IVport:
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
70: -- -- -- -- -- -- -- --

**Comment by Quas7** [comment]
I solved the issue. The touch sensor of the display interferes with the i2c bus. Display and cam port share the same i2c interface. The signals on SDA look a bit short peaked from time to time maybe because the known clock stretching issue (http://www.hobbytronics.co.uk/raspberry-pi-i2c-clock-stretching).

Add to /boot/config.txt the line "disable_touchscreen=1" to remove the touch controller (FT5x06 controller) from the i2c bus. I2C-1 is clean.

Also, the baudrate of the i2c interface should not be slowed down to much as the iv board will not respond to commands as far as I tested.

The i2c bus problem is described in more detail here: http://www.optisimon.com/raspberrypi/touch/ft5406/2016/07/13/raspberry-pi-7-inch-touchscreen-hacking/
